### PR TITLE
Fibonacci

### PR DESCRIPTION
**Pull request by bartosz-gruszczyk**
Task 2 'fibonacci' solution

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: fibonacci
🏅 Granted 6 XP for @bartosz-gruszczyk

**Comment by ziobron** [comment]
⏱ 3 XP extra for @bartosz-gruszczyk for on-time delivery

File: homework/fibonacci/fibonacci.hpp [hunk 1]
```diff
@@ -1,11 +1,31 @@
...
+        int fibMinus1 = 1;
+        for(int i = 2; i <= sequence; ++i) {
+            fibN = fibMinus1 + fibMinus2;
+            fibMinus2 = fibMinus1;
+            fibMinus1 = fibN;
+        }
+    }    
+    else {
+        return sequence;
+    }
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:17]: else redundant.  Move this to the top and write like:

`if (sequence < 2) {
    return sequence;
}`

File: homework/fibonacci/fibonacci.hpp [hunk 2]
```diff
@@ -1,11 +1,31 @@
...
 
 int fibonacci_recursive(int sequence) {
-    // TODO: Your implementation goes here
+
+    if (sequence >= 3) {
+        return fibonacci_recursive(sequence - 1) + fibonacci_recursive(sequence - 2);
+    }
+    else if (sequence > 0) {
+        return 1;
+    }
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:28]: `else if` is redundant here.


If you return variable here, you don't need to use `else if`

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**